crates/test/macros/src/lib.rs

```diff
@@ -124,15 +124,17 @@ fn tests_runner_impl(args: TokenStream) -> parse::Result<TokenStream> {
                 ret
             }
 
-            fn test_list() -> impl Iterator<Item = (&'static str, &'static str, ::flipperzero_test::TestFn)> {
+            fn test_list() -> impl Iterator<Item = (&'static str, &'static str, ::flipperzero_test::TestFn)> + Clone {
                 let ret = ::core::iter::empty();
                 #( let ret = ret.chain(#test_lists); )*
                 ret
             }
 
             // Test runner entry point
-            fn main(_args: *mut u8) -> i32 {
-                match ::flipperzero_test::__macro_support::run_tests(test_count(), test_list()) {
+            fn main(args: *mut u8) -> i32 {
```

Comment:
Being a bit of a devils advocate here: `main` also, seems to be `unsafe` since the caller is again required to provide a valid pointer.

The same logic works here as with `parse`: since it is a generated function, it is accessible to nearby code which may, for example, call it with null.

Not to be too copy-pasty here, something like

```markdown
# Safety

This should only be called by Flipperzero OS
and never manually.
```

would be enough.

Reply:
Opened #98.